Problem 4: TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences.


In [1]:
import numpy as np
import pandas as pd
from scipy import stats

In [3]:
data = pd.read_csv("Costomer+OrderForm.csv")

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Phillippines  300 non-null    object
 1   Indonesia     300 non-null    object
 2   Malta         300 non-null    object
 3   India         300 non-null    object
dtypes: object(4)
memory usage: 9.5+ KB


In [5]:
data.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [10]:

data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [11]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [12]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [13]:

data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [17]:
# Make a contingency table
table=np.array([[271,267,269,280],[29,33,31,20]])
table

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

### We assume a null hypothesis Ho : defective percentage is same among four centers.
### Alternate hypothesis H1: defective percentage of the form is different among all centers.

In [25]:
from scipy.stats import chi2_contingency

In [26]:
# Chi2 contengency independence test
chi2_contingency(table) #(Chi2 stats value, p_value, df, expected obsvations)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

### pvalue = 0.277, pvalue > 0.05 so we fail to reject the null hypothesis, we can say that we accept the null hypothesis. We conclude that the defectice percentage is same in all four centers. 